src/cleanfigure.m

```diff
@@ -347,104 +349,161 @@ function simplifyLine(meta, handle, targetResolution)
     yData = get(handle, 'YData');
     zData = get(handle, 'ZData');
 
-    if ~isempty(zData)
-        % TODO: 3d simplificattion of frontal 2d projection
-        return;
-    end
-    if isempty(xData) || isempty(yData)
+    if numel(xData) <= 2 || numel(xData) <= 2
         return;
     end
-    if numel(xData) <= 2
-        return;
+
+    % Check whether this is a 3D plot
+    if numel(zData) >1
+        is3D = true;
+    else
+        is3D = false;
     end
```

Comment:
As isAxis3D is only a 2 liner and i am actually unsure whether it is fully correct, I changed it to check for the elevation. If the elevation is 90 degree, then we have a 2D plot.